src/python/zquantum/core/circuit/_gate_test.py

```diff
@@ -1,7 +1,184 @@
 import unittest
 import os
 from ._gate import Gate
+from ._gateset import COMMON_GATES, UNIQUE_GATES
+from ._qubit import Qubit
+from sympy import Symbol
+import qiskit
+from math import pi
+
 
 class TestGate(unittest.TestCase):
-    def test_test(self):
-        pass
\ No newline at end of file
+    def setUp(self):
+        self.two_qubit_gates = ["CNOT", "CZ", "CPHASE", "SWAP"]
+        self.one_parameter_gates = ["PHASE", "Rx", "Ry", "Rz", "CPHASE"]
+
+    def create_gate(self, gate_name, params=None):
+        if gate_name in self.two_qubit_gates:
+            qubit_list = [Qubit(0), Qubit(1)]
+        else:
+            qubit_list = [Qubit(0)]
+        if params is None:
+            params = []
+            if gate_name in self.one_parameter_gates:
+                params = [1.0]
+        gate = Gate(gate_name, qubits=qubit_list, params=params)
+        return gate, qubit_list
+
+    def create_gate_with_symbolic_params(self, gate_name):
```

Comment:
Same comments as for `create_gate`

Reply:
Done in `01b1a27f3c57ca60d3707f7a8497e7a50a6322f7`